### PR TITLE
Add default favicon.ico file

### PR DESCRIPTION
**Pull request by opsyne**
mushroom

I thought this would be a fitting default favicon. It's taken from twitter's twemoji. It's licensed under CC BY 4.0, but the twemoji project has more lax guidelines for attribution:

> However, we consider the guide a bit onerous and as a project, will accept a mention in a project README or an 'About' section or footer on a website.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,3 +44,7 @@ If you want to contribute with code, open a pull request on GitHub or send a pat
 If you want to report an issue, open an issue on GitHub or contact us directly.
 
 Consider supporting the development on [Boosty](https://boosty.to/bouncepaw).
+
+## Assets
+
+The `static/favico.ico` file in this repository is the [mushroom emoji](https://github.com/twitter/twemoji/blob/54df6a1/assets/72x72/1f344.png) from Twitter's [twemoji project](https://twemoji.twitter.com/), and is licensed under the [Creative Commons CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)
```
Comment (bouncepaw) [inline README.md:50]: favicon -> favicon

And the point it missing at the end of the sentence.

**Review by bouncepaw — CHANGES_REQUESTED** [review]
Checked the favicon on dark and light theme on laptop and smartphone, looks great.

What if somebody wants to _not_ have a favicon? What do they do?

**Comment by opsyne** [comment]
> What if somebody wants to not have a favicon? What do they do?

Hmm, that's a good point. We could add a config option called `UseDefaultFavicon`, but that might complicate the route handling a little bit. Telling users to create a transparent 1x1 px png could also work, but I feel like that's a pretty bad solution from a UX standpoint.

I'm a fan of the config option, personally. But if anyone else has a better idea I'm all for it!

**Comment by handlerug** [comment]
How about bundling the favicon with Mycorrhiza then extracting it into `<wiki_path>/static/` on wiki creation? This way, people who don't want this icon can delete it. And we don't have to pollute the config file as well.

**Comment by bouncepaw** [comment]
> We could add a config option called `UseDefaultFavicon`

I avoid options generally. What happens if `UseDefaultFavicon=true`, but there is a `wiki/static/favicon.ico` provided by user?

> Telling users to create a transparent 1x1 px png could also work, but I feel like that's a pretty bad solution from a UX standpoint.

It is.

> How about bundling the favicon with Mycorrhiza then extracting it into `<wiki_path>/static/` on wiki creation?

That might work. Mycorrhiza creates a lot of files anyway.


----

Regarding the licensing of the icon. I just remembered we already have icons from other projects. Their sources are mentioned on `static/icon/README.md`, it makes more sense to have the licensing info for the shroom there as well.

**Comment by opsyne** [comment]
> How about bundling the favicon with Mycorrhiza then extracting it into <wiki_path>/static/ on wiki creation?

+1 from me as well

> Regarding the licensing of the icon. I just remembered we already have icons from other projects. Their sources are mentioned on static/icon/README.md, it makes more sense to have the licensing info for the shroom there as well.

Sounds good!

**Comment by opsyne** [comment]
Like I mentioned in the commit message, that solution might be a bit over-engineered.

**Comment by handlerug** [comment]
Looks completely fine to me! My only concern (just noticed) is that the icon is a PNG file, but the name is `favicon.ico` as if it's an ICO image file. That shouldn't be an issue (every mainstream browser will double-check the magic bytes anyway), but maybe @bouncepaw has a different opinion. (By the way, ICO files support embedding PNG-compressed images.)

File: files/files.go [hunk 2]
```diff
@@ -77,5 +83,41 @@ func PrepareWikiRoot() error {
 	paths.categoriesJSON = filepath.Join(cfg.WikiDir, "categories.json")
 	paths.interwikiJSON = FileInRoot("interwiki.json")
 
+	// Are we initializing the wiki for the first time?
+	if isFirstInit {
+		err := firstTimeInit()
+		if err != nil {
+			return err
```
Comment (handlerug) [inline files/files.go:90]: Though you may want to just log the error without taking down the program because those assets aren't critical at all.

**Comment by bouncepaw** [comment]
PNG is fine.

**Review by bouncepaw — APPROVED** [review]

**Pull request merged**